app/models/manageiq/providers/azure/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,64 @@
+class ManageIQ::Providers::Azure::CloudManager::EventCatcher::Stream
+  #
+  # Creates an event monitor
+  #
+  def initialize(client_id, client_key, azure_region, tenant_id)
+    @client_id         = client_id
+    @client_key        = client_key
+    @tenant_id         = tenant_id
+    @azure_region      = azure_region
+    @collecting_events = false
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  def each_batch
+    while @collecting_events
+
+      # Grab only events for the last minute if this is the first poll
+      startup_interval = (Time.current - 1.minute).httpdate
```

Comment:
this is only used for the first iteration of the loop.
how about inlining it or extracting a method?

you might even consider extracting the whole filter logic into their own methods and leave `each_batch` only concerned with yielding events